components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -763,15 +765,34 @@ internal sealed class SubscriptionWrapper : ISubscription {
                                 continue;
                             }
 
-                            var monitoredItemNotification = notification.Events[i]
-                                .ToMonitoredItemNotification(monitoredItem);
-                            if (message == null) {
-                                continue;
-                            }
-                            message.Notifications?.Add(monitoredItemNotification);
-
                             if (monitoredItem.Handle is MonitoredItemWrapper itemWrapper) {
                                 var pendingAlarmsOptions = itemWrapper?.EventTemplate?.PendingAlarms;
+                                var evFilter = (itemWrapper.Item.Filter as EventFilter);
```

Comment:
High risk for something being null here right? EventFilter... EventTypeIndex. Even itemWrapper above is used as null propagation.